client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -184,6 +184,18 @@ function mediaButton( editor ) {
 			let mediaHasCaption = false;
 			let captionNode = null;
 
+			// If image is deleted in image editor, we delete it in the post/page editor.
+			if ( media && media.status === 'deleted' ) {
+				captionNode = editor.dom.getParent( img, 'div.mceTemp' );
+				if ( captionNode ) {
+					editor.$( captionNode ).remove();
+				} else {
+					editor.$( img ).remove();
+				}
```

Comment:
looks like the action is the same here regardless of which branch we take so I wonder if we couldn't express this in a way which makes the conditional focus more on the element being operated on than the action?

```js
const imageNode = captionNode
	? captionNode
	: img;

editor.$( imageNode ).remove();
editor.nodeChanged();
```